### PR TITLE
Shop structs and named vars

### PR DESCRIPTION
**Pull request by Illeprih**
Struct for the list of shops. Named shop related variables

### CODE REVIEW

File: src/main/java/legend/game/combat/Bttl_800e.java [hunk 1]
```diff
@@ -6072,13 +6072,13 @@ public static void FUN_800ef28c(final long a0) {
...
       s0.itemSpPerMagicalHit_12e.set(stats.spPerMagicalHit_52.get());
       s0.mpPerMagicalHit_130.set(stats.mpPerMagicalHit_54.get());
-      s0.hpRegen_132.set(stats.hpRegen_56.get());
-      s0.mpRegen_134.set(stats.mpRegen_58.get());
-      s0.spRegen_136.set(stats.spRegen_5a.get());
-      s0.revive_138.set(stats._5c.get());
-      s0._13a.set(stats._5e.get());
-      s0._13c.set(stats._62.get());
-      s0._13e.set(stats._64.get());
+      s0.hpRegen_132.set(stats._56.get());
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/Bttl_800e.java:6075]: Hmm, what's going on with these?

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

File: src/main/java/legend/game/combat/Bttl_800e.java (hunk 1 above)
Comment (Illeprih) [reply src/main/java/legend/game/combat/Bttl_800e.java]: Apparently these values were also fixed in the battle obj. Fixed it, tested in battle and it now regenerates correctly.

File: src/main/java/legend/game/SItem.java [hunk 2]
```diff
@@ -462,8 +462,14 @@ private SItem() { }
   public static final Pointer<Renderable58> renderable_8011e0f4 = MEMORY.ref(4, 0x8011e0f4L, Pointer.deferred(4, Renderable58::new));
   public static final UnboundedArrayRef<MenuItemStruct04> menuItems_8011e0f8 = MEMORY.ref(4, 0x8011e0f8L, UnboundedArrayRef.of(0x4, MenuItemStruct04::new));
 
-  public static final UnsignedByteRef _8011e13c = MEMORY.ref(1, 0x8011e13cL, UnsignedByteRef::new);
-  public static final Value _8011e13d = MEMORY.ref(1, 0x8011e13dL);
+  public static final UnsignedByteRef currentShopItemCount_8011e13c = MEMORY.ref(1, 0x8011e13cL, UnsignedByteRef::new);
+  /**
```
Comment (LordMonoxide) [inline src/main/java/legend/game/SItem.java:466]: Reformat javadocs and add missing *'s at the start of the lines (applies to multiple multi-line javadocs in this PR)

File: src/main/java/legend/game/SItem.java [hunk 3]
```diff
@@ -2953,7 +2959,7 @@ public static void renderInventoryMenu(final long selectedOption, final long a1,
       v1 = submapNames_8011c108.get((int)_800bd808.get()).deref();
     } else {
       //LAB_80101ec0
-      v1 = worldMapNames_8011c1ec.get((int)_800bf0b0.get()).deref();
+      v1 = worldMapNames_8011c1ec.get((int) continentIndex_800bf0b0.get()).deref();
```
Comment (LordMonoxide) [inline src/main/java/legend/game/SItem.java:2962]: No spaces after cast

File: src/main/java/legend/game/SMap.java [hunk 4]
```diff
@@ -30,44 +30,7 @@
...
-import legend.game.types.Struct20;
-import legend.game.types.Struct34;
-import legend.game.types.Struct54;
-import legend.game.types.SubmapEncounterData_04;
-import legend.game.types.TmdAnimationFile;
-import legend.game.types.TmdExtension;
-import legend.game.types.UnknownStruct;
-import legend.game.types.UnknownStruct2;
-import legend.game.types.WeirdTimHeader;
+import legend.game.types.*;
```
Comment (LordMonoxide) [inline src/main/java/legend/game/SMap.java:33]: Please reconfigure your IDE to not use * imports

File: src/main/java/legend/game/combat/types/BattleStructEF4/AdditionExtra04.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+
+/**
+ * One for each ally and enemy
+ */
+public class AdditionExtra04 implements MemoryRef {
+    private final Value ref;
+
+    /**
+     <ul>
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/types/BattleStructEF4/AdditionExtra04.java:14]: Same as javadoc formatting I mentioned before

File: src/main/java/legend/game/combat/types/BattleStructEF4/SpecialEffects20.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+import legend.core.memory.*;
+import legend.core.memory.types.*;
+
+/**
+ * One for each ally and enemy
+ */
+public class SpecialEffects20 implements MemoryRef {
+    private final Value ref;
+    /**
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/types/BattleStructEF4/SpecialEffects20.java:11]: Same as javadoc formatting I mentioned before

File: src/main/java/legend/game/combat/types/BattleStructEF4/SpecialEffects20.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+     <li>0x10 Dragoon </li>
+     <li>0x20 D-Attack </li>
+     <li>0x40 Magic </li>
+     <li>0x80 Special </li>
+     <ul>
+     */
+    public final UnsignedByteRef menuBlockFlag_18;
+    /**
+     * Each effect has two bits for up to 3 turns
+     <ul>
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/types/BattleStructEF4/SpecialEffects20.java:26]: Same as javadoc formatting I mentioned before

File: src/main/java/legend/game/combat/types/BattleStructEF4/Status04.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+ * One for each ally and enemy
+ */
+public class Status04 implements MemoryRef {
+    private final Value ref;
+    /**
+     * Bunch of flags. Somehow monster/ally turn matters.
+     */
+    public final UnsignedByteRef statusEffect_00;
+    /**
+    Does not decrease on all statuses
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/types/BattleStructEF4/Status04.java:16]: Same as javadoc formatting I mentioned before

File: src/main/java/legend/game/combat/types/BattleStructEf4.java [hunk 9]
```diff
@@ -1,17 +1,41 @@
...
+
   public final ArrayRef<IntRef> _180;
 
+  /**
+   * Number of times pressed during the Dragoon addition.
+   */
+  public final UnsignedByteRef dAttackValue_280;
+
+  /**
+   <ul>
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/types/BattleStructEf4.java:20]: Same as javadoc formatting I mentioned before

File: src/main/java/legend/game/types/ScriptState.java [hunk 10]
```diff
@@ -71,8 +64,26 @@ public static <T extends MemoryRef> Class<ScriptState<T>> classFor(final Class<T
    * </ul>
    */
   public final UnsignedIntRef ui_60; // Note: also contained in previous array
+  /**
+   * Battle Menu flag - controls which options you can choose from.
+   <ul>
```
Comment (LordMonoxide) [inline src/main/java/legend/game/types/ScriptState.java:69]: Same as javadoc formatting I mentioned before

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

**Pull request merged**